src/install/install.zig

```diff
@@ -6658,10 +6670,60 @@ pub const PackageManager = struct {
         }
     };
 
+    pub fn enqueueGitForCheckout(
```

Comment:
hardmode: package with dependencies that depend on two different git SHAs

Reply:
We do have an existing test for the fresh installation case:
https://github.com/oven-sh/bun/blob/1fa7c1f79e1a0e665fc5a02764bdda724363d007/test/cli/install/bun-install.test.ts#L2620-L2630